src/main/java/com/lambdaschool/oktafoundation/controllers/ReportController.java

```diff
@@ -0,0 +1,205 @@
+package com.lambdaschool.oktafoundation.controllers;
+
+
+import com.lambdaschool.oktafoundation.models.Club;
+import com.lambdaschool.oktafoundation.models.MemberReactions;
+import com.lambdaschool.oktafoundation.repository.ClubRepository;
+import com.lambdaschool.oktafoundation.repository.MemberReactionRepository;
+import com.lambdaschool.oktafoundation.services.ClubService;
+import com.lambdaschool.oktafoundation.services.MemberReactionService;
+import com.lambdaschool.oktafoundation.views.ClubActivityPosivity;
+import com.lambdaschool.oktafoundation.views.MemberPosivity;
+import org.hibernate.transform.AliasToEntityMapResultTransformer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.EntityManager;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/report")
+public class ReportController {
+
+    private HashMap<String,Integer> emojimap;
+
+    public ReportController(){
+        emojimap = new HashMap<>();
+        emojimap.put("1F601", 2);
+        emojimap.put("1F642", 1);
+        emojimap.put("1F610", 0);
+        emojimap.put("1F641", -1);
+        emojimap.put("1F61E", -2);
+    }
+
+    @Autowired
+    private MemberReactionService memberReactionService;
+
+    @Autowired
+    private ClubRepository clubRepository;
+
+    @Autowired
+    private EntityManager entityManager;
+
+    /* The endpoints shall serve data analysis of the member reactions
+    with respect to club/activity/member.
+     */
+
+    // Currently we have the simple feature of calculating the avgs of positivity value of all members
+    // and avgs of the received posivity regarding activities in a club.
+
+
+    @PreAuthorize("hasAnyRole('ADMIN','CD')")
```

Comment:
return members' avg emovalue against all activities.
If cid = 0, we run it against all clubs, otherwise, restrict to a single club